contrib/mesos/pkg/scheduler/plugin.go

```diff
@@ -302,16 +313,15 @@ func (k *kubeScheduler) Schedule(pod *api.Pod, unused algorithm.NodeLister) (str
 	}
 }
 
-// Call ScheduleFunc and subtract some resources, returning the name of the machine the task is scheduled on
-func (k *kubeScheduler) doSchedule(task *podtask.T, err error) (string, error) {
+// doSchedule schedules the given task and returns the machine the task is scheduled on
+// or an error if the scheduling failed.
+func (k *kubeScheduler) doSchedule(task *podtask.T) (string, error) {
 	var offer offers.Perishable
+	var err error
+
 	if task.HasAcceptedOffer() {
 		// verify that the offer is still on the table
-		offerId := task.GetOfferId()
-		if offer, ok := k.api.offers().Get(offerId); ok && !offer.HasExpired() {
-			// skip tasks that have already have assigned offers
-			offer = task.Offer
-		} else {
+		if offer, ok := k.api.offers().Get(task.GetOfferId()); !ok || offer.HasExpired() {
```

Comment:
good catch! Indeed, the var is being shadowed! (saw this in the old impl. and didn't add the fix here, my bad!)